### ISSUE TITLE
Crash on startup due to no playlist image

### ISSUE THREAD

**Issue by Drakulix** [xou816/spot#54]
Hey there,

I installed Spot via Flathub on Arch Linux.
I am running sway as a compositor with some gnome-services started manually.
Everything was working fine until after login, where the program crashed.

Launching now gives me the following stacktrace:
`flatpak run --env=RUST_BACKTRACE=full dev.alextren.Spot`

```
home
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/app/backend/api_models.rs:329:50
stack backtrace:
   0:     0x5569fae73230 - <unknown>
   1:     0x5569fae96cdc - <unknown>
   2:     0x5569fae6be12 - <unknown>
   3:     0x5569fae754d5 - <unknown>
   4:     0x5569fae7502a - <unknown>
   5:     0x5569fae75c71 - <unknown>
   6:     0x5569fae75787 - <unknown>
   7:     0x5569fae736ec - <unknown>
   8:     0x5569fae75719 - <unknown>
   9:     0x5569fae944f1 - <unknown>
  10:     0x5569fae9443d - <unknown>
  11:     0x5569fa68443e - <unknown>
  12:     0x5569fa6d58f7 - <unknown>
  13:     0x5569fa6b4b73 - <unknown>
  14:     0x5569fa737371 - <unknown>
  15:     0x5569fa72e87e - <unknown>
  16:     0x5569fa71f90c - <unknown>
  17:     0x5569fa710bae - <unknown>
  18:     0x5569fa671d64 - <unknown>
  19:     0x5569fae46785 - <unknown>
  20:     0x5569fae4809a - <unknown>
  21:     0x7f46befac40b - g_main_context_dispatch
  22:     0x7f46befac6b8 - <unknown>
  23:     0x7f46befac783 - g_main_context_iteration
  24:     0x7f46bf1d246d - g_application_run
  25:     0x5569fa6e085e - <unknown>
  26:     0x5569fa694408 - <unknown>
  27:     0x7f46befad535 - g_main_context_invoke_full
  28:     0x5569fa763589 - <unknown>
  29:     0x5569fa68d873 - <unknown>
  30:     0x5569fa656919 - <unknown>
  31:     0x5569fae76097 - <unknown>
  32:     0x5569fa7639c2 - main
  33:     0x7f46bec13062 - __libc_start_main
  34:     0x5569fa62613e - <unknown>
  35:                0x0 - <unknown>
  ```
  
Which appears to be an issue related to fetching the [image of a playlist](https://github.com/xou816/spot/blob/master/src/app/backend/api_models.rs#L329)?
  
Not sure how to deal with this.

**Comment by Drakulix** [comment]
Fixed it. Had an empty playlist (which can be a thing in spotify, but I am not sure how I ended up with that), which therefor had no image.
The easy work-around would probably be to filter out such playlists (they do not provide much value), but it might be preferable to have some default art for the initialization of the `PlaylistDescription`.

Anyway, after deleting the playlist and clearing the cache of Spot, it works.

**Comment by xou816** [comment]
Hi,
Thanks for the detailed report!
Right, I should really not assume there to be at least one image :/ I knew that unwrap() was bad and yet I wrote it, shame on me! I'll look into making the URLs in these models Options.

**Comment by alx242** [comment]
I have this issue as well...instant crash on startup. My backtrace looks the same as @Drakulix

**Comment by bobodrone** [comment]
> Anyway, after deleting the playlist and clearing the cache of Spot, it works.

Could you point me in the direction on how to clear the cache, please? I did try to delete the ~/.var/app/dev.alextren.Spot/cache but on restart it still dies. I am running the latest release 0.1.5.

**Comment by xou816** [comment]
I’ll push a fix ASAP, but in the meantime you can indeed try to clear the cache! The location you have is correct.

If the error persists, could you try removing the empty playlist from your account, the clear the cache? If you can’t, it might be still possible to fool the app: assuming you identify the playlist or albums that’s the culprit (we’re looking for something without a cover art), you could edit it’s cache entry to include some picture, or edit it out of the list of playlists/albums. If you have any details about the playlist or album involved I’ll be able to give you more detailed instructions. 

Alternatively, you could try the latest build here : https://github.com/xou816/spot/suites/2062684952/artifacts/41648291 (x86 only)

sorry for the  trouble!

**Comment by Drakulix** [comment]
> > Anyway, after deleting the playlist and clearing the cache of Spot, it works.
> 
> 
> 
> Could you point me in the direction on how to clear the cache, please? I did try to delete the ~/.var/app/dev.alextren.Spot/cache but on restart it still dies. I am running the latest release 0.1.5.

Yes that is the right folder (at least for the flatpak variant).

Are you sure you deleted all problematic playlist before?

All I can tell you is that this worked for me

**Comment by bobodrone** [comment]
> Are you sure you deleted all problematic playlist before?

Thanx! I deleted maybe 10 empty lists but actually forgot one. :)
Now it seam to work!

**Comment by marcusti** [comment]
Same issue here on Pop!_OS 20.10. 

The suggested workaround works for me:
- Delete empty playlist from Spotify
- Delete everything under `~/.var/app/dev.alextren.Spot/cache/`

**Comment by alx242** [comment]
What does empty playlist actually mean? I have a bunch of playlists that uses local data which isn't present on Spotify. Would these be considered empty as well? I really don't want to delete them either...

**Comment by xou816** [comment]
It's not so much the playlist being empty that's the issue, it's the playlist having no image, which can happen for multiple reasons, one being an empty playlist. If your playlist has an image, you should be fine :) please note that local tracks are not supported however (filtered out).

For those of you having this issue, could you try and see if using this snapshot build https://github.com/xou816/spot/suites/2062684952/artifacts/41648291 solves the issue? (avoids the crash at least) :) Thanks!

**Comment by alx242** [comment]
> For those of you having this issue, could you try and see if using this snapshot build https://github.com/xou816/spot/suites/2062684952/artifacts/41648291 solves the issue? (avoids the crash at least) :) Thanks!

Worked like a charm :D

**Comment by JBBailly** [comment]
I had the same issue. I deleted an old playlist without image (I did it on open.spotify.com, as the official linux client crashed when I tried on it ...).
It fixed the problem too :+1:  
Thanks !

**Comment by xou816** [comment]
fixed by 58a1db78e8c7cc8bbc54b635bde6214495f441da in 0.1.8 (unreleased)
